pkg/kubectl/cmd/set/set_env.go

```diff
@@ -215,7 +215,13 @@ func (o *EnvOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []stri
 	o.Resources = resources
 	o.Cmd = cmd
 
-	o.PrintFlags.Complete(o.DryRun)
+	if o.DryRun {
+		// TODO(juanvallejo): This can be cleaned up even further by creating
+		// a PrintFlags struct that binds the --dry-run flag, and whose
+		// ToPrinter method returns a printer that understands how to print
+		// this success message.
+		o.PrintFlags.Complete("%s (dry run)")
```

Comment:
@deads2k updated the Complete method to receive a string template adding the "dry run" bits to the message.